frontend/src/pages/warehouse/ModalAddWarehouse.js

```diff
@@ -0,0 +1,324 @@
+            .then(resp => resp.json())
+            .then(res => {
+                const optionsFromBack = res.map((i) => ({
+                    id: i.id,
+                    name: i.name,
+                    surname: i.surname,
+                    username: i.username
+                }));
+                setDispatcherOptions(optionsFromBack);
+            });
+    };
+
+    const onChangeState = (e) => {
+        e.length > 0 ?
+            setWarehouseDto(preState => ({
+                ...preState,
+                addressDto: {...preState.addressDto, state: {id: e[0].id, state: e[0].state}}
+            })) :
+            setWarehouseDto(preState => ({
+                ...preState,
+                addressDto: {...preState.addressDto, state: {id: '', state: ''}}
+            }));
+    };
+
+    const addDispatcher = (e) => {
+        if (e.length !== 0) {
+            const dispatcherId = e[0].id;
+            const isContains = dispatchersList.filter((dispatcher) => dispatcher.id === dispatcherId);
+            if (isContains.length === 0) {
+                e.map(dispatcher =>
+                    setDispatchersList(preState => ([
+                        ...preState, dispatcher
+                    ])));
+            }
+        }
+    };
+
+    const handleIdentifier = (e) => {
+        setWarehouseDto(preState => ({
+            ...preState,
+            identifier: e.target.value
+        }));
+    };
+
+    const handleType = (e) => {
+        setWarehouseDto(preState => ({
+            ...preState,
+            type: e
+        }));
+        setDropdownMenuName(e);
+    };
+
+    const handleCity = (e) => {
+        setWarehouseDto(preState => ({
+            ...preState,
+            addressDto: {...preState.addressDto, city: e.target.value}
+        }));
+    };
+
+    const handleLineOne = (e) => {
+        setWarehouseDto(preState => ({
+            ...preState,
+            addressDto: {...preState.addressDto, addressLine1: e.target.value}
+        }));
+    };
+
+    const handleLineTwo = (e) => {
+        setWarehouseDto(preState => ({
+            ...preState,
+            addressDto: {...preState.addressDto, addressLine2: e.target.value}
+        }));
+    };
+
+    const handleTotalCapacity = (e) => {
+        setWarehouseDto(preState => ({
+            ...preState,
+            totalCapacity: e.target.value
+        }));
+    };
+
+    const addWarehouseHandler = (e) => {
+        e.preventDefault();
+        const dispatchersId = dispatchersList.map(dispatcher => dispatcher.id);
+        const updateWarehouseDto = {...warehouseDto, dispatchersId: dispatchersId, customerId: props.currentCustomerId}
+        let validationResult = validateWarehouse(warehouseDto, dispatchersId, dropdownMenuName);
+        setErrors(preState => ({
+            ...preState,
+            validationErrors: validationResult,
+            serverErrors: ''
+        }));
+        if (validationResult.length === 0) {
```

Comment:
if (!validationResult.length) may be used. Read about boolean evaluation in js.

Reply:
cool, I'll take it